### PR TITLE
Implement Special Gifts and Add Affection Calculation to Scouting Logic

### PR DESCRIPTION
**Pull request by imann24**
https://app.asana.com/0/235862522711549/319039773705922

**JSON**
- Added Tub-Tub sprite to Sprites.json
- Added JSON file for gifts (currently only contains gift events)
- Added to tuning variables related to special gifts to Tuning.json
- Added message for dog returning with another dog to Languages.json

**Scripts**
- Added `GiftEventData` and `DogVoucherData` gifts
- Removed deprecated `HomeSlotData` references
- Updated `PPGiftController` to incorporate other special gift data
- Added cheat to `PPGiftController` to make dogs only return special gifts
- Added cheats to `ToolController` for maxing out affection for all dogs in scene and adopted all dogs

**Prefabs**
- Updated RedeemDisplay prefab and GiftReportUI prefab with new font, and scaled font better on RedeemDisplay

**Sprites**
- Added placeholder sprite for Tub-Tub

### CODE REVIEW

File: Assets/Scripts/Data/Testing/GiftEventsTest.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * Author(s): Isaiah Mann
+ * Description: [to be added]
```
Comment (gracebarsny) [inline Assets/Scripts/Data/Testing/GiftEventsTest.cs]: Description needed.

File: Assets/Scripts/Exceptions/DogNotFoundException.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿/*
+ * Author(s): Isaiah Mann
+ * Description: Exception thrown when a dog cannot be located from a search or lookup
+ * Usage: [no notes]
+ */
+
+public class DogNotFoundException : PPException
```
Comment (gracebarsny) [inline Assets/Scripts/Exceptions/DogNotFoundException.cs]: Nice. I've been wanting something like this.

File: Assets/Scripts/Data/PickupPup/Currency/DogVoucherData.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	LanguageDatabase languages
+	{
+		get
+		{
+			return LanguageDatabase.GetInstance;
+		}
+	}
+
```
Comment (imann24) [inline Assets/Scripts/Data/PickupPup/Currency/DogVoucherData.cs]: @gracebarsny My policy is only putting public accessors inside that region.

**Comment by imann24** [comment]
@gracebarsny For testing, there's a cheat `bool' on `PPGiftController` for `alwaysReturnSpecialGift.` Also added cheats for maxing out affection of all adopted dogs and adopted all dogs. Those combined with the already existing cheat to speed up time should be sufficient

File: Assets/Scripts/Data/PickupPup/Currency/DogVoucherData.cs (hunk 3 above)
Comment (gracebarsny) [inline Assets/Scripts/Data/PickupPup/Currency/DogVoucherData.cs]: Got it!

**Comment by gracebarsny** [comment]
@imann24 Changes look great! About to test for functionality.

**Pull request merged**